### DISCUSSION TITLE
Surprising behavior with `beta` tags on `develop`

### DISCUSSION THREAD

**Discussion by dennisdoomen** [adamralph/minver#826] [General]
I was trying to tag a commit on our `develop` branch, but were surprised that MinVer 4.2.0 wasn't generating versions based on that tag. See below debug output. 

```
MinVer: Ignoring { Commit: f80fce1, Tag: '9.7.0-beta.1',      Version: 9.7.0-beta.1,      Height:   32 }.
MinVer: Ignoring { Commit: 781d8ca, Tag: '9.7.0-beta.2',      Version: 9.7.0-beta.2,      Height:   26 }.
MinVer: Using    { Commit: bc8c27d, Tag: '9.7.0-develop.0',   Version: 9.7.0-develop.0,   Height:  242 }.
MinVer: The calculated version 9.7.0-develop.0.242 satisfies the minimum major minor 0.0.
MinVer: Calculated version 9.7.0-develop.0.242.
9.7.0-develop.0.242
```

So instead of getting a version like `9.7.0-beta.2.x` where `x` is the height, it ignored the `beta` tags and based itself on the much older `9.7.0-develop.0` tag. 

What's even more surprising is that according to GH, the number of commits between `9.7.0-develop.0` and the current commit is much higher.

![image](https://user-images.githubusercontent.com/572734/197951639-d640e40b-704d-4db4-86f4-b17b7c6acb09.png)

In fact, before we tagged commit `f80fce1` as `9.7.0-beta.1`, MinVer was generating `9.7.0-develop.0.305`. 

Are we doing something wrong, are we misunderstanding MinVer or is this a bug?

**Comment by maxcherednik** [comment]
As far as I understand it works as described here: https://github.com/adamralph/minver#what-if-the-history-diverges-and-more-than-one-tag-or-root-commit-is-found

**Comment by adamralph** [comment]
@dennisdoomen if you switch the [verbosity](https://github.com/adamralph/minver#can-i-get-log-output-to-see-how-minver-calculates-the-version) to `diagnostic`/`trace`, it should be clearer what's going on.

**Comment by dennisdoomen** [comment]
That output came from a `diagnostic` build, but will try `trace` as well.

**Reply by adamralph** [reply]
`diagnostic`/`trace` are the same thing. The former is for the `MinVer` package and the latter is for the `minver-cli` package. See https://github.com/adamralph/minver#can-i-get-log-output-to-see-how-minver-calculates-the-version

The output you posted looks like `detailed`/`debug`, which is less verbose.
